src/buffers/disk/leveldb_buffer.rs

```diff
@@ -239,16 +244,28 @@ impl Reader {
 
             self.delete_offset = new_offset;
 
-            self.db.compact(&Key(0), &Key(self.delete_offset));
-
             let size_deleted = self.unacked_sizes.drain(..num_to_delete).sum();
             self.current_size.fetch_sub(size_deleted, Ordering::Relaxed);
+
+            self.uncompacted_size += size_deleted;
+            if self.uncompacted_size as f64 > self.max_size as f64 * MAX_UNCOMPACTED {
+                self.compact();
+            }
```

Comment:
My only real concern here is that this can cause us to use significantly more disk space than `max_size`, which is already a somewhat confusing configuration option for users.

Maybe instead of `uncompacted_size` being in addition to `max_size`, we split `max_size` and use ~90% for normal data and ~10% for uncompacted data (or 95/5 or something like that)? This will reduce how much we store when the buffer is full but should help us stay within `max_size` for total disk use and avoid the situation where we compact on every write.

Reply:
> Maybe instead of uncompacted_size being in addition to max_size, we split max_size and use ~90% for normal data and ~10% for uncompacted data

True, that will make more sense.  I'll play a bit with 20%, 10%, and 5%, and see what's happening.

Reply:
10% will do fine. I'm curious what the benchmarks will show though.

Reply:
They still show the same improvement, so let's go with 10%.